notebooks/Binary Classification with the UCI Credit Card Payment Default Dataset.ipynb

```diff
@@ -0,0 +1,599 @@
+    "import pandas as pd\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline\n",
+    "\n",
+    "# Data processing\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "# Models\n",
+    "import lightgbm as lgb\n",
+    "from sklearn.calibration import CalibratedClassifierCV\n",
+    "\n",
+    "# Fairlearn models and utils\n",
+    "from fairlearn.postprocessing import ThresholdOptimizer\n",
+    "from fairlearn.reductions import GridSearch, EqualizedOdds\n",
+    "from fairlearn.widget import FairlearnDashboard\n",
+    "\n",
+    "# Metrics\n",
+    "from fairlearn.metrics import group_selection_rate, group_zero_one_loss, group_recall_score, \\\n",
+    "                              metric_by_group, group_roc_auc_score\n",
+    "from sklearn.metrics import balanced_accuracy_score, accuracy_score, recall_score, roc_auc_score"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# The UCI Credit Card Payment Default Dataset\n",
+    "\n",
+    "The UCI datasets contains on 30,000 clients and their credit card transactions at a bank in Taiwan. In addition to static client features, the data contains the history of credit card bill payments between April and September 2005, as well as the balance limit of the client's credit card. The target is whether the client will default on a card payment in the following month, October 2005. One can imagine that a model trained on this data can be used in practice to determine whether a client is eligible for another product offering such as an auto loan. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load the data\n",
+    "data_url = \"http://archive.ics.uci.edu/ml/machine-learning-databases/00350/default%20of%20credit%20card%20clients.xls\"\n",
+    "dataset = pd.read_excel(io=data_url, header=1).drop(columns=['ID']).rename(columns={'PAY_0':'PAY_1'})\n",
+    "dataset.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Dataset columns:\n",
+    "\n",
+    "* `LIMIT_BAL`: credit card limit, will be replaced by a biased feature\n",
+    "* `SEX, EDUCATION, MARRIAGE, AGE`: client demographic features\n",
+    "* `BILL_AMT[1-6]`: amount on bill statement for the April-September months\n",
+    "* `PAY_AMT[1-6]`: payment amount for the April-September months\n",
+    "* `default payment next month`: target, whether the customer defaulted the following month"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Sensitive feature\n",
+    "A = dataset[\"SEX\"]\n",
+    "A_str = A.map({ 2:\"female\", 1:\"male\"})\n",
+    "# Target\n",
+    "Y = dataset[\"default payment next month\"]\n",
+    "categorical_features = ['EDUCATION', 'MARRIAGE','PAY_1', 'PAY_2', 'PAY_3', 'PAY_4', 'PAY_5', 'PAY_6']\n",
```

Comment:
Might want an extra comment about flagging columns as categorical. Personally, I'd leave a blank line between the three different parts of the cell, but that's even more minor :-)